pkg/kubelet/container/runtime.go

```diff
@@ -305,6 +306,18 @@ type PodStatus struct {
 	SandboxStatuses []*runtimeapi.PodSandboxStatus
 }
 
+// ContainerResources represents the Resources allocated to the running container.
+type ContainerResources struct {
+	// CPU capacity reserved for the container (cpu.shares)
+	CPURequest *resource.Quantity
+	// CPU limit enforced on the container (cpu.cfs_quota_us)
+	CPULimit *resource.Quantity
+	// Memory capaacity reserved for the container
+	MemoryRequest *resource.Quantity
```

Comment:
this field is unset when on cgroups v1, so it does not seem possible for the runtime to populate it.  we will need a follow-up on this struct to clean it up for v2 host.  in particular, we probably also need to figure out how to handle memory.min/high as part of memory qos feature as that is tied to a throttling factor kubelet side.  we can handle this post alpha, but if you can keep a //TODO here linking to the FeatureGate name that would help.

the runtime is not told what the memory request 
this field is unset in a cgroups v1 host, correct?

the runtime is not mapping memory request to any value because the kubelet is enforcing memory qos on v1 with the oom score adjust.

I think

Reply:
Yes, will do. I'll track it with issues. 
@liubog2008 Are you still up for taking on the cgroupv2 task? :)

Reply:
@vinaykul  I'm sorry for my delay. I have no enough time to work on it because my work scope changes. Maybe it's OK in April or May.

Reply:
@liubog2008 No worries. Apr/May would be fine - this is definitely something we can tackle pre-beta. I just cursorily went through the cgroup v2 documentation and this gives us request/limit symmetry for memory as we have for CPU. The pod resize code is structured to add memory requests to the mix this with fairly minimal changes but would need some work for v1/v2. So it looks like a decent slice of work E2E. I'll create an issue for this and assign it to you if that's ok :)

Reply:
Ugh! My cgroupv2 support is missing setting Request.Memory. Also, we may need to extend CRI if it is not in unified map. Tracking issue https://github.com/kubernetes/kubernetes/issues/114202